src/test/java/org/folio/innreach/controller/InnReachTransactionControllerTest.java

```diff
@@ -1807,6 +1807,41 @@ void cancelPatronHold_when_TransactionIsOnHoldOrTransfer_and_RequestIsClosed(Tra
 
     assertNotNull(updatedTransaction);
     assertEquals(TransactionStateEnum.BORROWING_SITE_CANCEL, updatedTransaction.getState());
+//    assertEquals(null,updatedTransaction.getHold().getFolioHoldingId());
+//    assertEquals(null,updatedTransaction.getHold().getFolioLoanId());
+//    assertEquals(null,updatedTransaction.getHold().getFolioInstanceId());
+//    assertEquals(null,updatedTransaction.getHold().getFolioItemId());
+
+    verify(circulationClient, never()).updateRequest(eq(PRE_POPULATED_PATRON_HOLD_REQUEST_ID), any());
+    verify(actionNotifier).reportCancelItemHold(any());
+    verify(innReachClient).postInnReachApi(any(), anyString(), anyString(), anyString());
+  }
+
+  @ParameterizedTest
+  @EnumSource(names = {"PATRON_HOLD", "TRANSFER"})
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-inn-reach-transaction.sql",
+  })
+  void cancelPatronHold_when_TransactionIsOnHoldOrTransfer_and_RequestIsClosed_withNoVirtualRecord(TransactionState state) {
+    mockFindRequest(PRE_POPULATED_PATRON_HOLD_REQUEST_ID, CLOSED_CANCELLED);
+
+    modifyTransactionState(PRE_POPULATED_PATRON_HOLD_TRANSACTION_ID, state);
+    var cancelPatronHold = createCancelTransactionHold();
+
+    var responseEntity = testRestTemplate.postForEntity(
+      PATRON_HOLD_CANCEL_ENDPOINT, cancelPatronHold, InnReachTransactionDTO.class,
+      PRE_POPULATED_PATRON_HOLD_TRANSACTION_ID);
+
+    assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
+    var updatedTransaction = responseEntity.getBody();
+
+    assertNotNull(updatedTransaction);
+    assertEquals(TransactionStateEnum.BORROWING_SITE_CANCEL, updatedTransaction.getState());
+//    assertEquals(null,updatedTransaction.getHold().getFolioHoldingId());
+//    assertEquals(null,updatedTransaction.getHold().getFolioLoanId());
+//    assertEquals(null,updatedTransaction.getHold().getFolioInstanceId());
+//    assertEquals(null,updatedTransaction.getHold().getFolioItemId());
```

Comment:
Same as above.

Reply:
These can be un-commented now, I have made the require change and committed back.